### PR TITLE
fix(src/containers/article): add beforeunload eventListener

### PR DESCRIPTION
**Pull request by liruchen32**
# Issue
[[會員制] 閱讀文章篇數、閱讀時間串接](https://app.asana.com/0/1203699264568808/1206008027632075/f) defects

# Notice

# Dependency

### CODE REVIEW

File: src/containers/Article.js [hunk 1]
```diff
@@ -160,25 +165,33 @@ class Article extends PureComponent {
...
+    }
+  }
+
   componentDidMount() {
-    const { fontLevel, changeFontLevel, slugToFetch, isAuthed } = this.props
+    const { fontLevel, changeFontLevel, slugToFetch } = this.props
 
     // Fetch the full post
     this.fetchAFullPostWithCatch(slugToFetch)
-    if (isAuthed) {
```
Comment (Aylie-Chou) [inline src/containers/Article.js:168]: Question: do we have any other method to register those event handler only when user has authed ?

**Review by Aylie-Chou — CHANGES_REQUESTED** [review]
Good replacement to use `pagehide` instead of `beforeunload`
Just have the concern about the extra loading caused by new added event handler when user is not authed.
Please have a look ~

File: src/containers/Article.js (hunk 1 above)
Comment (liruchen32) [reply src/containers/Article.js:168]: 有 debounce & throttle 應該不會造成太大的負擔

**Review by Aylie-Chou — APPROVED** [review]
LGTM
The reason of how `isAuthed` influence event register could be found out in following survey.

**Pull request merged**